src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+package com.aap.gitst;
+
+import static com.aap.gitst.RepoProperties.PROP_CA;
+import static com.aap.gitst.RepoProperties.PROP_CATHREADS;
+import static com.aap.gitst.RepoProperties.PROP_DEFAULT_CATHREADS;
+import static com.aap.gitst.RepoProperties.PROP_DEFAULT_IGNORE;
+import static com.aap.gitst.RepoProperties.PROP_DEFAULT_USER_PATTERN;
+import static com.aap.gitst.RepoProperties.PROP_IGNORE;
+import static com.aap.gitst.RepoProperties.PROP_PASSWORD;
+import static com.aap.gitst.RepoProperties.PROP_URL;
+import static com.aap.gitst.RepoProperties.PROP_USER;
+import static com.aap.gitst.RepoProperties.PROP_USER_PATTERN;
+import static com.starbase.starteam.ServerConfiguration.PROTOCOL_TCP_IP_SOCKETS_XML;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+import com.starbase.starteam.CheckinManager;
+import com.starbase.starteam.CheckinOptions;
+import com.starbase.starteam.CheckoutManager;
+import com.starbase.starteam.CheckoutOptions;
+import com.starbase.starteam.Folder;
+import com.starbase.starteam.Item;
+import com.starbase.starteam.LogonException;
+import com.starbase.starteam.Project;
+import com.starbase.starteam.Server;
+import com.starbase.starteam.ServerInfo;
+import com.starbase.starteam.StarTeamFinder;
+import com.starbase.starteam.StarTeamURL;
+import com.starbase.starteam.User;
+import com.starbase.starteam.View;
+import com.starbase.starteam.ViewConfiguration;
+import com.starbase.starteam.vts.comm.NetMonitor;
+import com.starbase.util.OLEDate;
+
+/**
+ * @author Andrey Pavlenko
+ */
+public class Repo implements AutoCloseable {
+    private static final String FOOTER_START = "----------------------------------- StarTeam -----------------------------------";
+    private static final String FOOTER_END = "--------------------------------------------------------------------------------";
+    public static final String LS = System.getProperty("line.separator");
+    public static final String DATE_FORMAT = "dd.MM.yy HH:mm:ss";
+    private final RepoProperties _repoProperties;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "_repoProperties" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00116)